platform/security/platform-security-core-api/src/main/java/ddf/security/common/audit/SecurityLogger.java

```diff
@@ -13,235 +13,275 @@
 import org.apache.cxf.phase.PhaseInterceptorChain;
-import org.apache.cxf.security.SecurityContext;
-import org.apache.cxf.service.model.EndpointInfo;
 import org.apache.cxf.transport.http.AbstractHTTPDestination;
-import org.apache.cxf.ws.security.tokenstore.SecurityToken;
-import org.apache.cxf.ws.security.tokenstore.TokenStore;
-import org.apache.cxf.ws.security.tokenstore.TokenStoreFactory;
-import org.apache.wss4j.common.principal.SAMLTokenPrincipal;
-import org.apache.wss4j.dom.engine.WSSecurityEngineResult;
-import org.apache.wss4j.dom.handler.WSHandlerConstants;
-import org.apache.wss4j.dom.handler.WSHandlerResult;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.w3c.dom.DOMImplementation;
-import org.w3c.dom.Document;
-import org.w3c.dom.Element;
-import org.w3c.dom.Node;
-import org.w3c.dom.ls.DOMImplementationLS;
-import org.w3c.dom.ls.LSSerializer;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.util.Supplier;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.subject.Subject;
 
 import ddf.security.SecurityConstants;
+import ddf.security.SubjectUtils;
 
 /**
  * Class that contains utility methods for logging common security messages.
  */
 public final class SecurityLogger {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SecurityConstants.SECURITY_LOGGER);
+
+    private static final Logger LOGGER = LogManager.getLogger(SecurityConstants.SECURITY_LOGGER);
+
+    private static final String NO_USER = "UNKNOWN";
 
     private SecurityLogger() {
 
     }
 
-    private static String requestIpAndPortMessage(Message message) {
+    private static String getUser(Subject subject) {
+        String user;
+        try {
+            if (subject == null) {
+                subject = SecurityUtils.getSubject();
+            }
+            user = SubjectUtils.getName(subject, NO_USER);
+        } catch (Exception e) {
+            user = NO_USER;
+        }
+
+        return user;
+    }
+
+    private static String requestIpAndPortAndUserMessage(Message message) {
+        return requestIpAndPortAndUserMessage(null, message);
+    }
+
+    private static String requestIpAndPortAndUserMessage(Subject subject, Message message) {
+        String user = getUser(subject);
         if (message != null) {
             HttpServletRequest servletRequest = (HttpServletRequest) message.get(
                     AbstractHTTPDestination.HTTP_REQUEST);
             // pull out the ip and port of the incoming connection so we know
             // who is trying to get access
             if (servletRequest != null) {
-                return " Request IP: " + servletRequest.getRemoteAddr() + ", Port: "
-                        + servletRequest.getRemotePort();
+                return "Subject: " + user + " Request IP: " + servletRequest.getRemoteAddr()
```

Comment:
If performance is a concern, maybe we should consider creating a `StringBuilder` before calling the method and passing it in? Not a big deal, just a suggestion.